packages/core/src/main.ts

```diff
@@ -75,6 +76,8 @@ export function getStarknet(
   }
   const lastConnectedStore = storageFactoryImplementation("gsw-last")
 
+  bootstrapMetamaskBridge();
```

Comment:
since this method will be async, if we place it here,  it may fall into a racing condition when checking if the metamask installed and snap is enabled , but the AvailableWallets has already provided/display to client

IMHO, i would suggest inject this method into `getAvailableWallets, getPreAuthorizedWallets, getDiscoveryWallets, getLastConnectedWallet`

and since bootstrapMetamaskBridge will inject the wallet object into window object, there is a checking in the method to avoid duplicate process, means if we trigger second time, it wont needed to reimport everything

how does it sound like to you?

e.g 
```
getAvailableWallets: async (options = {}) => {
      await bootstrapMetamaskBridge();

      const availableWallets = scanObjectForWallets(
        windowObject,
        isWalletObject,
      )
      return pipe<StarknetWindowObject[]>(
        (_) => filterBy(_, options),
        (_) => sortBy(_, options.sort),
      )(availableWallets)
  },
```

Reply:
As we spoke, `bootstrapMetamaskBridge` cannot be called in order to keep them synchronous.

Reply:
resolved